### PR TITLE
fix: parsing ddl error

### LINKED ISSUE #293: Display DDL parsing errors onChange in schema.sql tab
DDL schema parser returns an error like this:
```{
    "message": "Expected \")\", \",\", \"--\", \"/*\", \"AUTO_INCREMENT\", \"CHARACTER\", \"COLLATE\", \"COLUMN_FORMAT\", \"COMMENT\", \"DEFAULT\", \"KEY\", \"NOT NULL\", \"NULL\", \"ON\", \"PRIMARY\", \"REFERENCES\", \"STORAGE\", \"UNIQUE\", or [ \\t\\n\\r] but \"\\\"\" found.",
    "expected": [
        {
            "type": "class",
            "parts": [
                " ",
                "\t",
                "\n",
                "\r"
            ],
            "inverted": false,
            "ignoreCase": false
        },
        {
            "type": "literal",
            "text": "/*",
            "ignoreCase": false
        },
        {
            "type": "literal",
            "text": "--",
            "ignoreCase": false
        },
        {
            "type": "literal",
            "text": "DEFAULT",
            "ignoreCase": true
        },
        {
            "type": "class",
            "parts": [
                " ",
                "\t",
                "\n",
                "\r"
            ],
            "inverted": false,
            "ignoreCase": false
        },
        {
            "type": "literal",
            "text": "/*",
            "ignoreCase": false
        },
        {
            "type": "literal",
            "text": "--",
            "ignoreCase": false
        },
        {
            "type": "class",
            "parts": [
                " ",
                "\t",
                "\n",
                "\r"
            ],
            "inverted": false,
            "ignoreCase": false
        },
        {
            "type": "literal",
            "text": "/*",
            "ignoreCase": false
        },
        {
            "type": "literal",
            "text": "--",
            "ignoreCase": false
        },
        {
            "type": "literal",
            "text": "NOT NULL",
            "ignoreCase": true
        },
        {
            "type": "literal",
            "text": "NULL",
            "ignoreCase": true
        },
        {
            "type": "literal",
            "text": "DEFAULT",
            "ignoreCase": true
        },
        {
            "type": "literal",
            "text": "AUTO_INCREMENT",
            "ignoreCase": true
        },
        {
            "type": "literal",
            "text": "UNIQUE",
            "ignoreCase": true
        },
        {
            "type": "literal",
            "text": "PRIMARY",
            "ignoreCase": true
        },
        {
            "type": "class",
            "parts": [
                " ",
                "\t",
                "\n",
                "\r"
            ],
            "inverted": false,
            "ignoreCase": false
        },
        {
            "type": "literal",
            "text": "/*",
            "ignoreCase": false
        },
        {
            "type": "literal",
            "text": "--",
            "ignoreCase": false
        },
        {
            "type": "literal",
            "text": "KEY",
            "ignoreCase": true
        },
        {
            "type": "literal",
            "text": "COMMENT",
            "ignoreCase": true
        },
        {
            "type": "literal",
            "text": "COLLATE",
            "ignoreCase": true
        },
        {
            "type": "literal",
            "text": "COLUMN_FORMAT",
            "ignoreCase": true
        },
        {
            "type": "literal",
            "text": "STORAGE",
            "ignoreCase": true
        },
        {
            "type": "literal",
            "text": "REFERENCES",
            "ignoreCase": true
        },
        {
            "type": "literal",
            "text": "ON",
            "ignoreCase": true
        },
        {
            "type": "literal",
            "text": "CHARACTER",
            "ignoreCase": true
        },
        {
            "type": "class",
            "parts": [
                " ",
                "\t",
                "\n",
                "\r"
            ],
            "inverted": false,
            "ignoreCase": false
        },
        {
            "type": "literal",
            "text": "/*",
            "ignoreCase": false
        },
        {
            "type": "literal",
            "text": "--",
            "ignoreCase": false
        },
        {
            "type": "literal",
            "text": ",",
            "ignoreCase": false
        },
        {
            "type": "class",
            "parts": [
                " ",
                "\t",
                "\n",
                "\r"
            ],
            "inverted": false,
            "ignoreCase": false
        },
        {
            "type": "literal",
            "text": "/*",
            "ignoreCase": false
        },
        {
            "type": "literal",
            "text": "--",
            "ignoreCase": false
        },
        {
            "type": "literal",
            "text": ")",
            "ignoreCase": false
        }
    ],
    "found": "\"",
    "location": {
        "start": {
            "offset": 101,
            "line": 5,
            "column": 5
        },
        "end": {
            "offset": 102,
            "line": 5,
            "column": 6
        }
    },
    "name": "SyntaxError"
}```

in console. We can display `message` every time user types text, or with some delay.

### PR DESCRIPTION
**Pull request by Kevin101Zhang**
Glyphs margins to showcase a better error message to the user. The Glyph is either red or green. Green indicating it has not caught an errors and red indicating it has found an error. When an error is found it will come with a hoverable error message as well as the line it was caught on.

Procedure:
Onload of the schema page the glyphs will mount and check if there is an error based off the users previously written schema or their locally stored code.

If the user has an error: It will highlight the line the error was caught on with a red glyph and a hoverable message.
If the user does not have an error it will remain on line 1 with a green glyph. Onhover it will tell the user there are no errors.

If the user navigates away from the schema tab with an error it will prompt the user with a Alert that there was an error and to please return back to the schema tab and correct the schema.

The Glyph functionality with run on a debounced event of 500ms based off the users onChange typing activity on the schema page. When triggered it will prettify the users schema and will attempt to generate types in a try/catch. If it hits the error then it will relay the line number the error was caught on and the err message which the glyph will represent.

The Glyph only takes place on schema.sql as for indexLogic we have monaco's support for intellisense/catching errors in the JS env.

https://www.loom.com/share/78d0c6435115470597b9e8448976fda4?sid=fba10322-2ded-4b14-8178-8a5b51143148

### CODE REVIEW

**Review by darunrs — COMMENTED** [review]
Looks good in general. There's some comments I have on it. But, there's some sort of linting thing that applied on the code, which messed around with spaces and tabbings and what not, which made this pretty tough to read through. 

It's better to do any visual refactoring individually rather than applying it broadly as it bloats the PR. 

Also, could you put the recorded video and a more detailed PR Description too? 

Otherwise, once that's all addressed, we can get it merged.

File: frontend/src/components/Editor/ResizableLayoutEditor.jsx [hunk 1]
```diff
@@ -68,17 +59,30 @@ const ResizableEditor = ({
...
+          options={{
+            wordWrap: "on",
+            minimap: { enabled: false },
+            folding: false,
+            lineNumberMinChars: 3,
+            scrollBeyondLastLine: false,
+            automaticLayout: true,
+            formatOnPaste: true,
+            definitionLinkOpensInPeek: true,
+            // glyphMargin: true,
```
[the remark this answers is not in the document]
Comment (Kevin101Zhang) [reply frontend/src/components/Editor/ResizableLayoutEditor.jsx:83]: yes intended to have a glyph matching the error only in our sql file

File: frontend/src/utils/pgSchemaTypeGen.js [hunk 2]
```diff
@@ -15,104 +15,108 @@ export class PgSchemaTypeGen {
...
-		}
 
-		const tsTypes = this.generateTypeScriptDefinitions(dbSchema);
-    console.log(`Types successfully generated`);
-		return tsTypes;
+			const tsTypes = this.generateTypeScriptDefinitions(dbSchema);
+			console.log(`Types successfully generated`);
+			return tsTypes;
+
+		} catch (error) {
```
[the remark this answers is not in the document]
Comment (Kevin101Zhang) [reply frontend/src/utils/pgSchemaTypeGen.js:114]: Line 56: **const schemaSyntaxTree = this.parser.astify(sqlSchema, { database: "Postgresql" });**
depending on the schema the user has may not always astify properly. If there is ever an error, either unsupported or invalid syntax we catch it and display in on a glyph event.

**Comment by darunrs** [comment]
Nice video! Responded to one comment. 

In the video, I noticed the last error you made, the red box showed up at the top instead of on the line. Was that a bug or a specific scenario? Curious how it behaves in these two conditions:

- Make an error on one line, then another on a second line. Does it show one and after resolution, switch to showing the second error? Or shows both?
- Two errors on the same line, how does it work?

I think we're better off without the green box for no error. I feel like that distracts from the actual coding since while developing, you swing between a schema with an error and one without. It's better to have a clean looking view of the environment when nothing is wrong.

**Pull request closed without merging**